zeppelin-server/src/main/java/org/apache/zeppelin/rest/LoginRestApi.java

```diff
@@ -52,6 +67,106 @@ public LoginRestApi() {
   }
 
 
+  @GET
+  @ZeppelinApi
+  public Response getLogin(@Context HttpHeaders headers) {
+    JsonResponse response = null;
+    if (isKnoxSSOEnabled()) {
+      KnoxJwtRealm knoxJwtRealm = getJTWRealm();
+      Cookie cookie = headers.getCookies().get(knoxJwtRealm.getCookieName());
+      if (cookie != null && cookie.getValue() != null) {
+        Subject currentUser = org.apache.shiro.SecurityUtils.getSubject();
+        if (!currentUser.isAuthenticated()) {
+          JWTAuthenticationToken token = new JWTAuthenticationToken(null, cookie.getValue());
+          response = procedeToLogin(currentUser, token);
+        }
+      }
+      if (response == null) {
+        Map<String, String> data = new HashMap<>();
+        data.put("redirectURL", constructKnoxUrl(knoxJwtRealm, knoxJwtRealm.getLogin()));
+        response = new JsonResponse(Status.OK, "", data);
+      }
+      return response.build();
+    }
+    return new JsonResponse(Status.FORBIDDEN).build();
+  }
+
+  private KnoxJwtRealm getJTWRealm() {
+    Collection realmsList = SecurityUtils.getRealmsList();
+    if (realmsList != null) {
+      for (Iterator<Realm> iterator = realmsList.iterator(); iterator.hasNext(); ) {
+        Realm realm = iterator.next();
+        String name = realm.getClass().getName();
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("RealmClass.getName: " + name);
+        }
+        if (name.equals("org.apache.zeppelin.realm.jwt.KnoxJwtRealm")) {
+          return (KnoxJwtRealm) realm;
+        }
+      }
+    }
+    return null;
+  }
+
+  private boolean isKnoxSSOEnabled() {
+    Collection realmsList = SecurityUtils.getRealmsList();
+    if (realmsList != null) {
+      for (Iterator<Realm> iterator = realmsList.iterator(); iterator.hasNext(); ) {
+        Realm realm = iterator.next();
+        String name = realm.getClass().getName();
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("RealmClass.getName: " + name);
```

Comment:
why not just LOG.debug()? it doesn't look sensitive

Reply:
Sure, yes, removed.